framework/DataObjects/XDataSet.py

```diff
@@ -1045,12 +1044,15 @@ def _setScalingFactors(self):
     for var in self._allvars:
       # if not a float or int, don't scale it
       # TODO this check is pretty convoluted; there's probably a better way to figure out the type of the variable
-      first = self._data.groupby(var).first()[var].item(0)
-      if (not isinstance(first,(float,int))) or np.isnan(first):# or self._data[var].isnull().all():
-        continue
-      mean = float(self._data[var].mean())
-      scale = float(self._data[var].std())
-      self._scaleFactors[var] = (mean,scale)
+      #first = self._data.groupby(var).first()[var].item(0)
```

Comment:
I suggest to leave a comment to this line so we know it may cause performance degradation